snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedFhirValueSetExpander.java

```diff
@@ -0,0 +1,203 @@
+			.sortBy(!CompareUtils.isEmpty(termFilter) ? SearchIndexResourceRequest.SCORE : SearchResourceRequest.Sort.fieldAsc("id"));
+
+		
+		if (!valueSet.hasCompose()) {
+			// do nothing, search all concepts
+		} else {
+			final ValueSetComposeComponent compose = valueSet.getCompose();
+			final ConceptSetComponent firstInclude = compose.getIncludeFirstRep();
+			final ConceptSetFilterComponent firstFilter = firstInclude.getFilterFirstRep();
+			
+			if ("constraint".equals(firstFilter.getProperty()) && FilterOperator.EQUAL.equals(firstFilter.getOp())) {
+				// Filter concepts by ECL
+				req.filterByQuery(firstFilter.getValue());
+			} else if ("constraint".equals(firstFilter.getProperty()) && FilterOperator.ISA.equals(firstFilter.getOp())) {
+				// Filter concepts by ancestor
+				req.filterByAncestor(firstFilter.getValue());
+			} else if ("concept".equals(firstFilter.getProperty()) && FilterOperator.IN.equals(firstFilter.getOp())) {
+				req.filterByQuery("^" + firstFilter.getValue());
+			} else {
+				throw new IllegalStateException("Unsupported implicit value set compose definition: " + compose);
+			}
+		}
+		
+		final Concepts concepts = req.buildAsync().execute(context);
+		
+		final ValueSet.ValueSetExpansionComponent expansion = new ValueSet.ValueSetExpansionComponent()
+				.setIdentifier(valueSet.getId())
+				.setTimestampElement(FhirModelHelpers.toDateTimeElement(new Date()))
+				.setTotal(concepts.getTotal());
+
+		expansion.addExtension(FhirValueSetExpander.EXTENSION_AFTER_PROPERTY_URL, new StringType(concepts.getSearchAfter()));
+		
+		final String baseUrl = valueSet.getUserString("baseUrl");
+		
+		for (Concept concept : concepts) {
+			var contains = new ValueSet.ValueSetExpansionContainsComponent()
+				.setCode(concept.getId())
+				.setSystem(baseUrl)
+				.setDisplay(concept.getTerm());
+			
+			if (parameters.getIncludeDesignations() != null && parameters.getIncludeDesignations().getValue()) {
+				includeDesignations(baseUrl, concept, contains);
+			}
+			
+			expansion.addContains(contains);
+		}
+		
+		return valueSet.setExpansion(expansion);
+	}
+
+	private void includeDesignations(final String baseUrl, Concept concept, ValueSetExpansionContainsComponent contains) {
+		/*
+		 * XXX: We create multiple tooling-independent representations in the general
+		 * concept representation, these need to be collapsed back into a single
+		 * internal instance.
+		 */
+		final List<SnomedDescription> snomedDescriptions = concept.getDescriptions()
+			.stream()
+			.map(d -> (SnomedDescription) d.getInternalDescription())
+			.distinct()
+			.toList();
+			
+		for (final SnomedDescription snomedDescription : snomedDescriptions) {
+
+			/* 
+			 * Convert language reference set ID, acceptability and description type triples into "designation-use-context" extensions
+			 * https://confluence.ihtsdotools.org/display/FHIR/Designation+extension
```

Comment:
It would be nice to have automated tests for these use cases as well, both BCP-47 and the new Designation extension. Please schedule the implementation of a few test cases before the release.

Reply:
`designation-use-context` verification was added in https://github.com/b2ihealthcare/snow-owl/pull/1339/commits/9293d5e4ccdbef5a2229ef3eefb253549538b448.